source/DSCResources/DSC_CMUserDiscovery/DSC_CMUserDiscovery.psm1

```diff
@@ -0,0 +1,558 @@
+    {
+        throw $_
+    }
+    finally
+    {
+        Set-Location -Path "$env:temp"
+    }
+}
+
+<#
+    .SYNOPSIS
+        This will set the desired state.
+
+    .PARAMETER SiteCode
+        Specifies the site code for Configuration Manager site.
+
+    .PARAMETER Enabled
+        Specifies the enablement of the User Discovery method.
+
+    .PARAMETER EnableDeltaDiscovery
+        Indicates whether Configuration Manager discovers resources created or modified in AD DS
+        since the last discovery cycle. If you specify a value of $True for this parameter,
+        specify a value for the DeltaDiscoveryMins parameter.
+
+    .PARAMETER DeltaDiscoveryMins
+        Specifies the number of minutes for the delta discovery.
+
+    .PARAMETER ADContainers
+        Specifies an array of names of Active Directory containers to match to the discovery.
+
+    .PARAMETER ADContainersToInclude
+        Specifies an array of names of Active Directory containers to add to the discovery.
+
+    .PARAMETER ADContainersToExclude
+        Specifies an array of names of Active Directory containers to exclude to the discovery.
+
+    .PARAMETER ScheduleInterval
+        Specifies the time when the scheduled event recurs.
+
+    .PARAMETER ScheduleCount
+        Specifies how often the recur interval is run.
+#>
+function Test-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Boolean])]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SiteCode,
+
+        [Parameter(Mandatory = $true)]
+        [Boolean]
+        $Enabled,
+
+        [Parameter()]
+        [Boolean]
+        $EnableDeltaDiscovery,
+
+        [Parameter()]
+        [ValidateRange(1,60)]
+        [UInt32]
+        $DeltaDiscoveryMins,
+
+        [Parameter()]
+        [String[]]
+        $ADContainers,
+
+        [Parameter()]
+        [String[]]
+        $ADContainersToInclude,
+
+        [Parameter()]
+        [String[]]
+        $ADContainersToExclude,
+
+        [Parameter()]
+        [ValidateSet('None','Days','Hours','Minutes')]
+        [String]
+        $ScheduleInterval,
+
+        [Parameter()]
+        [UInt32]
+        $ScheduleCount
+    )
+
+    Import-ConfigMgrPowerShellModule -SiteCode $SiteCode
+    Set-Location -Path "$($SiteCode):\"
+    $state = Get-TargetResource -SiteCode $SiteCode -Enabled $Enabled
+    $result = $true
+
+    if ($Enabled -eq $true)
+    {
+        $testParams = @{
+            CurrentValues = $state
+            DesiredValues = $PSBoundParameters
+            ValuesToCheck = @('Enabled','EnableDeltaDiscovery','DeltaDiscoveryMins')
+        }
+
+        $result = Test-DscParameterState @testParams -TurnOffTypeChecking -Verbose
+
+        if ($PSBoundParameters.ContainsKey('ScheduleInterval'))
+        {
+            if ($ScheduleInterval -ne 'None' -and -not $PSBoundParameters.ContainsKey('ScheduleCount'))
+            {
+                Write-Verbose -Message $script:localizedData.IntervalCount
```

Comment:
thoughts on making this Write-Warning vs Verbose would also apply to System

Reply:
Changed and pushed